src/credentials/GroupDataProvider.h

```diff
@@ -141,6 +141,27 @@ class GroupDataProvider
         }
     };
 
+    /**
+     *  Interface to listen for changes in the Group info.
+     */
+    class Listener
```

Comment:
fully qualified name seems to be `chip::Credentials::Listener` which just by name seems very generic (some sort of "listen about credentials". Could we maybe make the name a bit more explicit, even the comment says "group info changes" so `GroupChangeListener` may be more appropriate

Reply:
The fully qualified name is `chip::Credentials::GroupDataProvider::Listener`. However, a prefix does improve readability here, so I have renamed the class as `GroupListener`.